contractcourt/chain_arbitrator.go

```diff
@@ -22,11 +23,18 @@ import (
 	"github.com/lightningnetwork/lnd/lnwallet"
 	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
 	"github.com/lightningnetwork/lnd/lnwire"
+	"golang.org/x/sync/errgroup"
 )
 
 // ErrChainArbExiting signals that the chain arbitrator is shutting down.
 var ErrChainArbExiting = errors.New("ChainArbitrator exiting")
 
+const (
+	// chainArbTimeout is the timeout for the chain arbitrator to start
+	// the channel arbitrators for each channel.
+	chainArbTimeout = 5 * time.Minute
```

Comment:
I think we should identify the exact line that is blocking and fix it properly, like buffering or using a new goroutine. Throwing a random timeout feels hacky.

Reply:
I'm not sure we can fix this in `tapd`. If the sweeper _requires_ additional instructions from the `AuxSweeper`, but the aux sweeper cannot start because `lnd` isn't fully started, then we can't just make this non-blocking. Otherwise we wouldn't be able to return the correct aux leaf information.

Reply:
So I think a good solution is probably 2 then, why trigger the `advanceState` when we do it later on in the `channelAttendant`. I agree that random timeouts might not be a good solution here. Will evaluate how big the change might be.